run_benchmarks.rb

```diff
@@ -303,6 +303,18 @@ def run_benchmarks(ruby:, ruby_description:, name_filters:, out_path:, pre_init:
     args.executables[name] = executable.shellsplit
   end
 
+  opts.on("--chruby=NAME::VERSION OPTIONS", "ruby version under chruby and options to be benchmarked") do |v|
+    name, version = v.split("::", 2)
+    if version.nil?
+      version = name # allow skipping `NAME::`
+    end
+    version, *options = version.shellsplit
+    unless executable = ["/opt/rubies/#{version}/bin/ruby", "#{ENV["HOME"]}/.rubies/#{version}/bin/ruby"].find { |path| File.executable?(path) }
```

Comment:
Ah, this is a good fix. I had tried a similar thing with running chruby, but it's hard to get the environment right.

This won't change the gem environment, though, right? So it would only work if the Rubies are both using the same built native extensions?

Reply:
It's https://github.com/k0kubun/yjit-bench/blob/9fc2cd39d2ddebdb58f0ae7965530390a225cc7f/run_benchmarks.rb#L263-L267
And that's only done if it's not the first `ruby` in PATH, I missed that.
So then the behavior seems:
* If it's the first Ruby in PATH use GEM_HOME/GEM_PATH as set by chruby or anything else
* Otherwise use the default gem home inside the prefix